### ISSUE TITLE
Uri.tryParse will make the host to be lowercase

### ISSUE THREAD

**Issue by lanshifu** [pichillilorenzo/flutter_inappwebview#1402]
- [1] I have read the [Get Started - Installation](https://inappwebview.dev/docs/get-started/installation/) section
- [1] I have read and done the [Get Started - Setup Android](https://inappwebview.dev/docs/get-started/setup-android/) section
- [1] I have read and done the [Get Started - Setup iOS](https://inappwebview.dev/docs/get-started/setup-ios/) section
- [1] I have already searched for the same problem

## Environment

| Technology           | Version       |
| -------------------- | ------------- |
| Flutter version      |        3.3.5       |
| Plugin version       |         6.1.6      |
| Android version      |               |
| iOS version          |               |
| Xcode version        |               |

## Description

the URI host is upCase（mqqopensdkapi://bizAgent），
but I received lowercase in _shouldOverrideUrlLoading callback,by the code `avigationAction.request.url`

the App could not be launch，because the host is not right。

then I found the source code is url_request.g.dart 

```
final instance = URLRequest(
      url: map['url'] != null ? Uri.tryParse(map['url']) : null,
```

Uri.tryParse will make the host to be lowercase，Could you please use annother way to parse the Uri，not change to lowercase？

## Steps to reproduce

1.test Code:

var url = 'mqqopensdkapi://bizAgent';
var parseResult = Uri.tryParse(url);

print('url==$url');
print('parseUrl=$parseResult');

2. see the log

I/flutter: url==mqqopensdkapi://biz**A**gent
I/flutter: parseUrl=mqqopensdkapi://biz**a**gent

the **A**gent expect to be upcase，but the result is lowerCase。

## Images   

![image](https://user-images.githubusercontent.com/20284668/197978319-887ceabb-c309-4809-9d8d-50e3ef4556b5.png)

**Comment by pichillilorenzo** [comment]
Ok, it seems that `Uri` is causing some problems, also on version `5.x.x`.
At this point, maybe I should implement something similar to `Uri` that maintains also the "raw string value" of it, for issues such as this one.

**Comment by pichillilorenzo** [comment]
Released new version `6.0.0-beta.10` with new type `WebUri` to replace the `Uri` dart core type.
With `WebUri`, you have access to the raw string value used by the Uri parser.
So, if you need the unmodified string, use `navigationAction.request.url.rawValue`.

For more details on how `WebUri` works, check the [WebUri Official docs](https://inappwebview.dev/docs/web-uri).

**Comment by lanshifu** [comment]
thanks a lot.
